core/src/main/java/org/infinispan/cache/impl/TypeConverterDelegatingAdvancedCache.java

```diff
@@ -744,4 +744,12 @@ public void replaceAll(BiFunction<? super K, ? super V, ? extends V> function) {
       return returned;
    }
 
+   @Override
+   public AdvancedCache<K, V> lockAs(Object lockOwner) {
+      AdvancedCache<K, V> returned = super.lockAs(lockOwner);
+      if (returned != this && returned instanceof TypeConverterDelegatingAdvancedCache) {
+         ((TypeConverterDelegatingAdvancedCache) returned).entryFactory = this.entryFactory;
```

Comment:
Sadly this is just because there isn't a test that has this exact scenario. I will add a 5 line test to `StoreAsBinaryTest` that will reproduce the NPE.